packages/pg-v5/lib/bastion.js

```diff
@@ -81,37 +83,54 @@ function getConfigs (db) {
 
 exports.getConfigs = getConfigs
 
-function sshTunnel (db, dbTunnelConfig, timeout) {
-  return new Promise((resolve, reject) => {
-    // if necessary to tunnel, setup a tunnel
-    // see also https://github.com/heroku/heroku/blob/master/lib/heroku/helpers/heroku_postgresql.rb#L53-L80
-    let timer = setTimeout(() => reject(new Error('Establishing a secure tunnel timed out')), timeout)
-    if (db.bastionKey) {
-      let tun = tunnel(dbTunnelConfig, (err, tnl) => {
-        if (err) {
-          debug(err)
-          reject(new Error('Unable to establish a secure tunnel to your database.'))
-        }
-        debug('Tunnel created')
-        clearTimeout(timer)
-        resolve(tnl)
-      })
-      tun.on('error', (err) => {
-        // we can't reject the promise here because we may already have resolved it
-        debug(err)
-        cli.exit(1, 'Secure tunnel to your database failed')
-      })
-    } else {
-      clearTimeout(timer)
-      resolve()
+class Timeout {
```

Comment:
i rewrote the timeout bits of the `sshTunnel` into a class so they were always cancelable